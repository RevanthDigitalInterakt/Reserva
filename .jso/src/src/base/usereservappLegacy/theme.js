  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
