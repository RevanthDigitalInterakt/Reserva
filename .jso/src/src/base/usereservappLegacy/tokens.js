  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
