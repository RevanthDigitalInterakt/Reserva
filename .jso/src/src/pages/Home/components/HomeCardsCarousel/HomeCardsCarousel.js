  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _toConsumableArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[3]);
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _HomeCard = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function HomeCardsCarousel(_ref) {
    var data = _ref.data;
    var scrollX = (0, _react.useRef)(new _reactNative.Animated.Value(0)).current;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[8]).Box, Object.assign({}, (0, _testProps.default)('cards_carrousel_container'), {
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.Animated.FlatList, Object.assign({}, (0, _testProps.default)('cards_carrousel_animated_flat_list'), {
        onScroll: _reactNative.Animated.event([{
          nativeEvent: {
            contentOffset: {
              x: scrollX
            }
          }
        }], {
          useNativeDriver: true
        }),
        horizontal: true,
        showsHorizontalScrollIndicator: false,
        data: data.items,
        snapToOffsets: (0, _toConsumableArray2.default)(Array(data.items.length)).map(function (_x, i) {
          return i * (_configDeviceSizes.default.DEVICE_WIDTH * 0.85 - 48) + (i - 1) * 48;
        }),
        keyExtractor: function keyExtractor(item) {
          return `card-home-${item.image.url}`;
        },
        snapToAlignment: "start",
        scrollEventThrottle: 16,
        decelerationRate: "fast",
        contentContainerStyle: _$$_REQUIRE(_dependencyMap[9]).styles.flatListContainer,
        bounces: false,
        renderItem: function renderItem(_ref2) {
          var _item$filters;
          var item = _ref2.item;
          return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_HomeCard.default, {
            imageUrl: item.image.url,
            reference: item.reference,
            reservaMini: item.reservaMini,
            orderBy: item.orderBy,
            filters: {
              priceFilter: item == null ? undefined : (_item$filters = item.filters) == null ? undefined : _item$filters.priceFilter
            }
          });
        }
      })), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[8]).Box, {
        height: 24,
        flexDirection: "row",
        alignSelf: "center",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.Animated.View, Object.assign({}, (0, _testProps.default)('com.usereserva:id/cards_carrousel_animated_view'), {
          style: [_$$_REQUIRE(_dependencyMap[9]).styles.slidingIndicatorStyle, {
            position: 'absolute',
            transform: [{
              translateX: _reactNative.Animated.divide(scrollX, _configDeviceSizes.default.DEVICE_WIDTH * 0.88 - 48).interpolate({
                inputRange: [0, 1],
                outputRange: [6, 25.8]
              })
            }]
          }]
        })), data.items.map(function (item) {
          return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
            style: _$$_REQUIRE(_dependencyMap[9]).styles.bulletsWrapper,
            children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
              style: _$$_REQUIRE(_dependencyMap[9]).styles.bullet
            })
          }, `cards-item-${item.image.url}`);
        })]
      })]
    }));
  }
  var _default = exports.default = HomeCardsCarousel;
