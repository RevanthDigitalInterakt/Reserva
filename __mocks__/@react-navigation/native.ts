const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
