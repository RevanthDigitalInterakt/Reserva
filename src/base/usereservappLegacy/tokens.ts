export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
