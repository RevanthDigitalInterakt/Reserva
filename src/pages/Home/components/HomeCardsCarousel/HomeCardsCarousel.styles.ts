import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  flatListContainer: {
    paddingLeft: 4,
    paddingRight: 4,
  },

  slidingIndicatorStyle: {
    backgroundColor: '#6F6F6F',
    width: 7,
    height: 7,
    alignSelf: 'center',
    zIndex: 2,
    borderRadius: 7,
  },

  bulletsWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 19,
  },

  bullet: {
    borderWidth: 1,
    width: 7,
    height: 7,
    borderRadius: 7,
    borderColor: '#6F6F6F',
  },
});
