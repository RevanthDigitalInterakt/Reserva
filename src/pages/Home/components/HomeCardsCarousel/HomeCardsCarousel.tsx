import React, { useRef } from 'react';
import { Animated, View } from 'react-native';
import { Box } from '@usereservaapp/reserva-ui';
import type { HomeCarouselOutput } from '../../../../base/graphql/generated';
import testProps from '../../../../utils/testProps';
import configDeviceSizes from '../../../../utils/configDeviceSizes';
import { styles } from './HomeCardsCarousel.styles';
import HomeCard from '../HomeCard';

interface IHomeCardsCarousel {
  data: HomeCarouselOutput;
}

function HomeCardsCarousel({ data }: IHomeCardsCarousel) {
  const scrollX = useRef(new Animated.Value(0)).current;

  return (
    <Box {...testProps('cards_carrousel_container')}>
      <Animated.FlatList
        {...testProps('cards_carrousel_animated_flat_list')}
        onScroll={Animated.event(
          [{ nativeEvent: { contentOffset: { x: scrollX } } }],
          { useNativeDriver: true },
        )}
        horizontal
        showsHorizontalScrollIndicator={false}
        data={data.items}
        snapToOffsets={[...Array(data.items.length)].map(
          (_x, i) => i * (configDeviceSizes.DEVICE_WIDTH * 0.85 - 48) + (i - 1) * 48,
        )}
        keyExtractor={(item) => `card-home-${item.image.url}`}
        snapToAlignment="start"
        scrollEventThrottle={16}
        decelerationRate="fast"
        contentContainerStyle={styles.flatListContainer}
        bounces={false}
        renderItem={({ item }) => (
          <HomeCard
            imageUrl={item.image.url}
            reference={item.reference}
            reservaMini={item.reservaMini}
            orderBy={item.orderBy}
          />
        )}
      />

      <Box height={24} flexDirection="row" alignSelf="center">
        <Animated.View
          {...testProps('com.usereserva:id/cards_carrousel_animated_view')}
          style={[
            styles.slidingIndicatorStyle,
            {
              position: 'absolute',
              transform: [
                {
                  translateX: Animated.divide(
                    scrollX,
                    configDeviceSizes.DEVICE_WIDTH * 0.88 - 48,
                  ).interpolate({
                    inputRange: [0, 1],
                    outputRange: [6, 25.8],
                  }),
                },
              ],
            },
          ]}
        />

        {data.items.map((item) => (
          <View key={`cards-item-${item.image.url}`} style={styles.bulletsWrapper}>
            <View style={styles.bullet} />
          </View>
        ))}
      </Box>
    </Box>
  );
}

export default HomeCardsCarousel;
