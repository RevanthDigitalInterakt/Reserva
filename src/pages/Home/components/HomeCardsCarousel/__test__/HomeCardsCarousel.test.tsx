import {
  render,
  screen,
} from '@testing-library/react-native';
import React from 'react';

import { ThemeProvider } from 'styled-components/native';
import type { HomeCarouselOutput } from '../../../../../base/graphql/generated';
import { theme } from '../../../../../base/usereservappLegacy/theme';
import HomeCardsCarousel from '../HomeCardsCarousel';

// MOCKS
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

const mockedCardCarousel = {
  id: 'MAIN',
  type: 'MAIN',
  showtime: 10,
  items: [{
    mkt: false,
    linkMktIn: null,
    reservaMini: false,
    reference: 'collection:2894',
    orderBy: 'RELEVANCIA',
    facets: [{ key: 'productClusterIds', value: '2894', __typename: 'ProductFacetOutput' }],
    image: {
      url: 'https://images.ctfassets.net/6jsfqc13oxv4/RYtXBnHIjoNdgmEvoO9Km/d06af9b5124eca415b7dfbf56fad7163/banner-home-app-liqui_para_todos.gif', title: 'Liqui para Todos Agosto 2023', height: 1400, width: 1080, __typename: 'HomeCarouselItemImageOutput',
    },
  }, {
    mkt: false,
    linkMktIn: null,
    reservaMini: false,
    reference: 'collection:2648',
    orderBy: 'RELEVANCIA',
    facets: [{ key: 'productClusterIds', value: '2648', __typename: 'ProductFacetOutput' }],
    image: {
      url: 'https://images.ctfassets.net/6jsfqc13oxv4/4TFpt2F0YWUdrb2lZ2SYak/09ad88e0d507be9079da04be0733bd05/banner-home-app-od_ate129.jpg', title: 'Camisetas at√© R$129', height: 1400, width: 1080, __typename: 'HomeCarouselItemImageOutput',
    },
  }, {
    mkt: false,
    linkMktIn: null,
    reservaMini: false,
    reference: 'collection:2772',
    orderBy: 'RELEVANCIA',
    facets: [{ key: 'productClusterIds', value: '2772', __typename: 'ProductFacetOutput' }],
    image: {
      url: 'https://images.ctfassets.net/6jsfqc13oxv4/2PZVKTj8T4Vkf0rGIw3NsK/2a8f32e7fe4e14ceaec20157001d328f/banner-home-app-ofertas_ate60.gif', title: 'Home 2772 Ofertas Reserva', height: 1400, width: 1080, __typename: 'HomeCarouselItemImageOutput',
    },
  }],
};

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <HomeCardsCarousel data={mockedCardCarousel as HomeCarouselOutput} />
  </ThemeProvider>
);

describe('HomeCardsCarousel', () => {
  beforeEach(async () => {
    jest.useFakeTimers({ legacyFakeTimers: true });
    render(TestingComponent);
  });

  it('should render properly', () => {
    const card = screen.getByTestId('com.usereserva:id/cards_carrousel_animated_flat_list');
    expect(card).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
