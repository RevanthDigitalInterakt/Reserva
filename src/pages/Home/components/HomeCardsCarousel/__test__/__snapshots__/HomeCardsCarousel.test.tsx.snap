// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCardsCarousel should match to snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/cards_carrousel_container"
  style={
    Array [
      Object {},
    ]
  }
  testID="com.usereserva:id/cards_carrousel_container"
>
  <RCTScrollView
    accessibilityLabel="com.usereserva:id/cards_carrousel_animated_flat_list"
    bounces={false}
    contentContainerStyle={
      Object {
        "paddingLeft": 4,
        "paddingRight": 4,
      }
    }
    data={
      Array [
        Object {
          "facets": Array [
            Object {
              "__typename": "ProductFacetOutput",
              "key": "productClusterIds",
              "value": "2894",
            },
          ],
          "image": Object {
            "__typename": "HomeCarouselItemImageOutput",
            "height": 1400,
            "title": "Liqui para Todos Agosto 2023",
            "url": "https://images.ctfassets.net/6jsfqc13oxv4/RYtXBnHIjoNdgmEvoO9Km/d06af9b5124eca415b7dfbf56fad7163/banner-home-app-liqui_para_todos.gif",
            "width": 1080,
          },
          "linkMktIn": null,
          "mkt": false,
          "orderBy": "RELEVANCIA",
          "reference": "collection:2894",
          "reservaMini": false,
        },
        Object {
          "facets": Array [
            Object {
              "__typename": "ProductFacetOutput",
              "key": "productClusterIds",
              "value": "2648",
            },
          ],
          "image": Object {
            "__typename": "HomeCarouselItemImageOutput",
            "height": 1400,
            "title": "Camisetas atÃ© R$129",
            "url": "https://images.ctfassets.net/6jsfqc13oxv4/4TFpt2F0YWUdrb2lZ2SYak/09ad88e0d507be9079da04be0733bd05/banner-home-app-od_ate129.jpg",
            "width": 1080,
          },
          "linkMktIn": null,
          "mkt": false,
          "orderBy": "RELEVANCIA",
          "reference": "collection:2648",
          "reservaMini": false,
        },
        Object {
          "facets": Array [
            Object {
              "__typename": "ProductFacetOutput",
              "key": "productClusterIds",
              "value": "2772",
            },
          ],
          "image": Object {
            "__typename": "HomeCarouselItemImageOutput",
            "height": 1400,
            "title": "Home 2772 Ofertas Reserva",
            "url": "https://images.ctfassets.net/6jsfqc13oxv4/2PZVKTj8T4Vkf0rGIw3NsK/2a8f32e7fe4e14ceaec20157001d328f/banner-home-app-ofertas_ate60.gif",
            "width": 1080,
          },
          "linkMktIn": null,
          "mkt": false,
          "orderBy": "RELEVANCIA",
          "reference": "collection:2772",
          "reservaMini": false,
        },
      ]
    }
    decelerationRate="fast"
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="start"
    snapToOffsets={
      Array [
        -48,
        589.5,
        1227,
      ]
    }
    stickyHeaderIndices={Array []}
    style={Object {}}
    testID="com.usereserva:id/cards_carrousel_animated_flat_list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/card_container_collection:2894"
          style={
            Array [
              Object {},
            ]
          }
          testID="com.usereserva:id/card_container_collection:2894"
        >
          <View
            alignItems="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_button_collection:2894"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="com.usereserva:id/card_button_collection:2894"
            >
              <Image
                height={280}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://images.ctfassets.net/6jsfqc13oxv4/RYtXBnHIjoNdgmEvoO9Km/d06af9b5124eca415b7dfbf56fad7163/banner-home-app-liqui_para_todos.gif",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 621.5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/card_container_collection:2648"
          style={
            Array [
              Object {},
            ]
          }
          testID="com.usereserva:id/card_container_collection:2648"
        >
          <View
            alignItems="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_button_collection:2648"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="com.usereserva:id/card_button_collection:2648"
            >
              <Image
                height={280}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://images.ctfassets.net/6jsfqc13oxv4/4TFpt2F0YWUdrb2lZ2SYak/09ad88e0d507be9079da04be0733bd05/banner-home-app-od_ate129.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 621.5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/card_container_collection:2772"
          style={
            Array [
              Object {},
            ]
          }
          testID="com.usereserva:id/card_container_collection:2772"
        >
          <View
            alignItems="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_button_collection:2772"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="com.usereserva:id/card_button_collection:2772"
            >
              <Image
                height={280}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://images.ctfassets.net/6jsfqc13oxv4/2PZVKTj8T4Vkf0rGIw3NsK/2a8f32e7fe4e14ceaec20157001d328f/banner-home-app-ofertas_ate60.gif",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 0,
                      "width": 621.5,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    alignSelf="center"
    flexDirection="row"
    height={24}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "height": 24,
        },
      ]
    }
  >
    <View
      accessibilityLabel="com.usereserva:id/cards_carrousel_animated_view"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#6F6F6F",
          "borderRadius": 7,
          "height": 7,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 6,
            },
          ],
          "width": 7,
          "zIndex": 2,
        }
      }
      testID="com.usereserva:id/cards_carrousel_animated_view"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 19,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#6F6F6F",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 7,
            "width": 7,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 19,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#6F6F6F",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 7,
            "width": 7,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 19,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#6F6F6F",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 7,
            "width": 7,
          }
        }
      />
    </View>
  </View>
</View>
`;
